simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/cli/ConsoleCommand.java

```diff
@@ -4,66 +4,77 @@
+
 /**
  * Contains list of console commands representing user actions applicable to certain station.
  * For each command there is a validation logic and mapping to {@link UserMessage}
  */
 public enum ConsoleCommand {
 
     PLUG {
+        private static final int EVSE_ID_INDEX = 0;
+        private static final int CONNECTOR_ID_INDEX = 1;
+
         @Override
         public UserMessage toUserMessage(List<String> args) {
             validateArgs(args);
-            return new Plug(Integer.valueOf(args.get(0)));
+            return new Plug(Integer.valueOf(args.get(EVSE_ID_INDEX)), Integer.valueOf(args.get(CONNECTOR_ID_INDEX)));
         }
 
         @Override
         void validateArgs(List<String> args) {
-            validateLength(args, 1);
+            validateLength(args, 2);
 
-            int connectorIdArgIndex = 0;
-            validateNumeric(connectorIdArgIndex, args.get(connectorIdArgIndex));
+            validateEvseAndConnector(args, EVSE_ID_INDEX, CONNECTOR_ID_INDEX);
         }
     },
 
     UNPLUG {
+        private static final int EVSE_ID_INDEX = 0;
+        private static final int CONNECTOR_ID_INDEX = 1;
+
         @Override
         public UserMessage toUserMessage(List<String> args) {
             validateArgs(args);
-            return new Unplug(Integer.valueOf(args.get(0)));
+            return new Unplug(Integer.valueOf(args.get(EVSE_ID_INDEX)), Integer.valueOf(args.get(CONNECTOR_ID_INDEX)));
         }
 
         @Override
         void validateArgs(List<String> args) {
-            validateLength(args, 1);
+            validateLength(args, 2);
 
-            int connectorIdArgIndex = 0;
-            validateNumeric(connectorIdArgIndex, args.get(connectorIdArgIndex));
+            validateEvseAndConnector(args, EVSE_ID_INDEX, CONNECTOR_ID_INDEX);
         }
     },
 
     AUTH {
+        private static final int RFID_INDEX = 0;
+        private static final int EVSE_ID_INDEX = 1;
+
         @Override
         public UserMessage toUserMessage(List<String> args) {
             validateArgs(args);
-            return new Authorize(args.get(0), Integer.valueOf(args.get(1)));
+            return new Authorize(args.get(RFID_INDEX), Integer.valueOf(args.get(EVSE_ID_INDEX)));
         }
 
         @Override
         void validateArgs(List<String> args) {
             validateLength(args, 2);
 
-            int authKeyArgIndex = 0;
-            int connectorIdArgIndex = 1;
-            validateIdentifierString(authKeyArgIndex, args.get(authKeyArgIndex));
-            validateNumeric(connectorIdArgIndex, args.get(connectorIdArgIndex));
+            validateIdentifierString(RFID_INDEX, args.get(RFID_INDEX));
+            validateNumeric(EVSE_ID_INDEX, args.get(EVSE_ID_INDEX));
+        }
+
+        private void validateIdentifierString(int index, String arg) {
+            boolean validTokenId = !arg.isEmpty() && arg.matches("([a-z]|[A-Z]|[0-9]|\\*|-|_|=|:|\\+|\\||@|\\.){0,36}");
```

Comment:
Consider moving regex to Pattern.compile private static final field.

Reply:
done